pegasus/config/unicorn.rb

```diff
@@ -7,3 +7,15 @@
 stdout_path pegasus_dir('log/unicorn_stdout.log')
 working_directory pegasus_dir
 #logger $log
+
+before_fork do |server, _worker|
+  # Quit the old unicorn process
+  old_pid = "#{server.config[:pid]}.oldbin"
+  if File.exist?(old_pid) && server.pid != old_pid
+    begin
+      Process.kill("QUIT", File.read(old_pid).to_i)
+    rescue Errno::ENOENT, Errno::ESRCH
+      # someone else did our job for us
+    end
+  end
+end
```

Comment:
after a bit of experimentation, managed to refactor into a `Proc` shared between both config files.